indico/modules/events/registration/wallets/apple.py

```diff
@@ -0,0 +1,144 @@
+# This file is part of Indico.
+# Copyright (C) 2002 - 2024 CERN
+#
+# Indico is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see the
+# LICENSE file for more details.
+
+import json
+import os
+import tempfile
+from pathlib import Path
+
+import requests
+from cryptography import x509
+from cryptography.hazmat.primitives import hashes, serialization
+from cryptography.hazmat.primitives.serialization import pkcs7
+from flask import current_app
+from wallet.models import Barcode, BarcodeFormat, EventTicket, Pass
+
+from indico.core import signals
+from indico.core.config import config
+from indico.core.logger import Logger
+from indico.util.i18n import _
+from indico.web.flask.util import url_for
+
+
+logger = Logger.get('events.registration.apple_wallet')
+
+
+class IndicoPass(Pass):
+    def add_file_from_url(self, name, url):
+        try:
+            response = requests.get(url)
+            self._files[name] = response.content
+        except (requests.HTTPError, requests.ConnectionError):
```

Comment:
IIRC there's a generic `requests.RequestException` that covers both (look at the inheritance of the requests exceptions to confirm this though)